integrationTests/mock/contracts/elrond.go

```diff
@@ -44,12 +52,71 @@ func (ec *ElrondContract) createContractFunctions() {
 	ec.AddHandler("userRole", ec.userRole)
 	ec.AddHandler("getTokenIdForErc20Address", ec.getTokenIdForErc20Address)
+	ec.AddHandler("wasTransferActionProposed", ec.wasTransferActionProposed)
+	ec.AddHandler("proposeMultiTransferEsdtBatch", ec.proposeMultiTransferEsdtBatch)
+	ec.AddHandler("getActionSignerCount", ec.getActionSignerCount)
+	ec.AddHandler("proposeEsdtSafeSetCurrentTransactionBatchStatus", ec.proposeEsdtSafeSetCurrentTransactionBatchStatus)
 }
 
 func (ec *ElrondContract) getCurrentTxBatch(caller string, value string, arguments ...string) ([][]byte, error) {
 	log.Debug("getCurrentTxBatch", "caller", caller, "value", value, "arguments", arguments)
 
-	return make([][]byte, 0), nil
+	return make([][]byte, 0), nil //ec.currentTxBatch
+}
+
+func (ec *ElrondContract) wasTransferActionProposed(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("wasTransferActionProposed", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	batchId, _ := strconv.ParseInt(arguments[0], 10, 64)
+	ret := byte(1) //we assume we have the action in map
+	if _, ok := ec.action_mapper[big.NewInt(batchId)]; !ok {
+		ret = 0
+	}
+	return [][]byte{{ret}}, nil
+}
+func (ec *ElrondContract) proposeMultiTransferEsdtBatch(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("proposeMultiTransferEsdtBatch ", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	var args []string
+	err := json.Unmarshal([]byte(arguments[0]), &args)
+	if err != nil {
+		log.Error("ElrondContract: Error unmarshal arguments", "error", err.Error())
+		return make([][]byte, 0), nil
+	}
+	buf, _ := base64.URLEncoding.DecodeString(args[0])
+	batchId := int64(buf[0])
+	//every 3rd elements starting from 2nd
+	//is empty string for separation so we skip any of them
+	for  i:=2;i<len(args); i+=3 {
+		token, _ := base64.URLEncoding.DecodeString(args[i])
+		amountInBytes, _ := base64.URLEncoding.DecodeString(args[i+1])
+		log.Debug("token: ", token, "amount: ", amountInBytes[0])
+	}
+	log.Debug(strings.Join(args, ":"))
+
+	ret := byte(1) //we assume we have the action in map
+	if _, ok := ec.action_mapper[big.NewInt(batchId)]; !ok {
+		ret = 0
+		ec.action_mapper[big.NewInt(batchId)] = big.NewInt(0)
+	}
+	return [][]byte{{ret}}, nil
+}
+func (ec *ElrondContract) proposeEsdtSafeSetCurrentTransactionBatchStatus(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("proposeEsdtSafeSetCurrentTransactionBatchStatus ", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+
+	ret := byte(1)
+	return [][]byte{{ret}}, nil
+}
+func (ec *ElrondContract) getActionSignerCount(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("getActionSignerCount", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	batchId := int64(-1)
+	_, err := fmt.Sscan(arguments[0], &batchId)
```

Comment:
we should have a conversion for this returning batch ID from byte slice. Check my newest PR for an idea.